### PR TITLE
Release v0.9.0 Review

### PR DESCRIPTION
**Pull request by Kovensky**

### CODE REVIEW

**Comment by ghedo** [comment]
In #1407 someone asked if we could keep release notes of previous releases in RELEASE_NOTES as well, but then I forgot about it. Since there wasn't any opposition, could you look into it?

File: RELEASE_NOTES [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+Changes
+-------
+
+Changes that may break users' config files have been annotated with a `(!)`.
+
+- Note: mpv is not compatible with Lua 5.3. Lua 5.1 or 5.2 is required.
+- The minimum required libass version is now 0.12.1 or newer.
+- The minimum required FFmpeg version is now 2.4.0 (equiv. Libav 11) or newer.
+- `(!)` The internal libmpg123 support was removed.
```
Comment (wm4) [inline RELEASE_NOTES]: Maybe mention that it was unused by default in the previous release, even if it was compiled-in.

Comment (wm4) [inline RELEASE_NOTES]: Also, this doesn't break user config-files, not even if they used ``--ad``.

File: RELEASE_NOTES [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  - `(!)` `scale-down` has been renamed to `dscale` and now has its own set of
+    config options (e.g. `dscale-radius`).
+  - `(!)` Scaler radius no longer defaults to `3` but to a preferred value
+    that may be different for each filter.
+  - The `scale-radius` option may now go down as low as `0.5`, which is the
+    value used by the `nearest` filter.
+  - `spline36` is the new `cscale` default for `opengl-hq`. This might break
+    setups that use `fbo-format=8` (or less). To work around it, leave
+    `fbo-format` as its default, or set to something higher than 8, or set
+    `cscale=bilinear`, the previous default.
```
Comment (wm4) [inline RELEASE_NOTES]: I don't think `fbo-format=8` was valid (not with that exact value). Also, I'm unsure how a different fbo-format could break it. At worst you'd lose some quality due to lack of precision?

File: RELEASE_NOTES [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+- `(!)` `vf_format` no longer converts video to YUYV if there is no parameter.
+  Video is now passed unchanged unless a format is specifically requested.
```
Comment (wm4) [inline RELEASE_NOTES]: Although it's a behavior change, I don't know if it's worth mentioning. YUYV is extremely obscure, even Apple is giving up on it.

File: RELEASE_NOTES [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+  `--colormatrix-output-range` and `--colormatrix-primaries` options have been
+  converted into `vf_format` suboptions. See commit 27715b7 and the manual for
+  details.
+- The `device` subption to `ao_wasapi` has been deprecated in favor of
+  `--audio-device`.
+- `--video-rotate` now allows 360 as an argument instead of stopping at 359.
+- Several improvements to `af_scaletempo`.
+- Options that have multiple options and also include a "yes" option now
+  default to that if specified with no arguments.
+- The default value of `--cache-default` is now 150000 (150 MB, ~146 MiB).
```
Comment (wm4) [inline RELEASE_NOTES]: I think it's actually 153 MB. (Fun, huh.)

File: RELEASE_NOTES [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+
+- mpv no longer saves position on files that can't be resumed (issue #1701).
+- (X11) Fix the player thinking the mouse has left the window in some WMs /
+  embeddings (issue #1672).
+- mpv no longer freezes on wayland when the compositor stops asking it to draw
+  itself (e.g. when minimized) (issue #249).
+- `.ac3` files are no longer rejected by `--audio-file-auto` (issue #1759).
+- `ao_wasapi` now automatically enables `exclusive` when passthrough is
+  attempted (issue #1742).
+- Attempt to fix flickering on Intel VAAPI drivers (issue #1765).
```
Comment (wm4) [inline RELEASE_NOTES]: This also might change performance characteristics big time.

Comment (wm4) [inline RELEASE_NOTES]: Also: this affects only ``--vo=opengl`` with ``--hwdec=vaapi``.

File: RELEASE_NOTES [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+  enters the VO from outside.
+- The new `--ytdl-params` option can be set to arguments that are always given
+  to `youtube-dl` invocations. There is no sanity checking, so invalid options
+  can prevent `mpv` from working with http URLs.
+- `vo_direct3d_shaders` now supports NV12 colorspace without using stretchrect,
+  but it seems no drivers actually support that.
+- (Client API) Added a `rescan_external_files` command, as requested on issue
+  #1586.
+- (Client API) If enabled, initializing the Client API will now also load the
+  user's `mpv.conf`.
```
Comment (wm4) [inline RELEASE_NOTES]: If you're adding client API changes, maybe also mention that other API change... or link to the api changes file.

Comment (wm4) [inline RELEASE_NOTES]: ("that other API change" = the recent commits)

File: RELEASE_NOTES (hunk 2 above)
Comment (Kovensky) [inline RELEASE_NOTES]: Well, that's what @haasn told me, though I might have misunderstood it ._.

File: RELEASE_NOTES (hunk 4 above)
Comment (Kovensky) [inline RELEASE_NOTES]: \mixed units/

File: RELEASE_NOTES (hunk 5 above)
Comment (Kovensky) [inline RELEASE_NOTES]: "performance characteristics" in what sense; does it make playback slower in some setups?

File: RELEASE_NOTES (hunk 2 above)
Comment (haasn) [inline RELEASE_NOTES]: In current master, chroma merging / upscaling happens before conversion to RGB, which means that for 10-bit videos, you get a 16-bit texture containing 10-bit values (ie. the signal is scaled to 2^-6), which are then clipped to 8 bit instead - and 2^-6 of 8 bit is 2 bit.

In other words, by using an 8-bit FBO with a nontrivial cscale, you are reducing the effective bit depth of the image to 2 bits, which goes quite a bit beyond mere “quality loss”.

For 16 bit FBOs, it comes out as ~10 bit precision, which is still a quality loss compared to 16 bit but one that isn't perceptible either way.

Anyway, the shader branch fixes this, by doing range expansion right away.

**Comment by Kovensky** [comment]
@ghedo I added the 0.8 release notes at the tail; should I add all of the other previous release notes as well?

**Comment by ghedo** [comment]
The guy asked for all the releases (to quote him "the same contents as shown under the github "release" summary"), though many old releases don't have `RELEASE_NOTES` so that may require additional work in converting from the GitHub release page. It's not really high priority, so if it's too much work you decide how many releases to add.

File: RELEASE_NOTES [hunk 7]
```diff
@@ -0,0 +1,370 @@
...
+- `youtube-dl` will no longer download video streams when video playback
+  is disabled with `--no-video`.
+
+New features
+------------
+
+- `vo_opengl` features:
+  - Added `ewa_ginseng`, `ewa_hanning`, `robidoux`, `robidouxsharp`,
+    `oversample` and `haasnsoft` scalers.
+  - There are now `ewa_lanczonsoft` and `ewa_lanczossharp` aliases to
```
Comment (wiiaboo) [inline RELEASE_NOTES]: typo here s/lanczonsoft/lanczossoft/

**Pull request closed without merging**